src/logic/SystemPermissions.swift

```diff
@@ -14,25 +14,11 @@ class SystemPermissions {
 
     static func screenRecordingIsGranted() -> Bool {
         if #available(OSX 10.15, *) {
-            return screenRecordingIsGranted_()
+            return CGPreflightScreenCaptureAccess()
         }
         return true
     }
 
-    // workaround: public API CGPreflightScreenCaptureAccess and private API SLSRequestScreenCaptureAccess exist, but
```

Comment:
Have you seen this? We were not using this API because it wasn't the best in the past.

Reply:
Yes, I read it, I will make it that it uses it for OSX 15+

Reply:
AltTab checks permissions continuously. The user could remove permissions at any point. We check for that on a repeating timer.

As the comment says, `CGPreflightScreenCaptureAccess` was returning a value which was not updated. It was a snapshot of the permissions at the time the app was launched, but didn't update after that initial snapshot.

Has this behavior changed in macOS 15? If it hasn't, then we should change other parts of the app doing the repeating timer checks. Either find another way to check there, or remove the check. But if we remove the check, we need another way to notice when the `screenshot()` and the AX calls start failing.

Reply:
I changed it to the canRecordScreen, this is updated during runtime